src/utils/payment-providers/stripe-provider.js

```diff
@@ -67,9 +67,17 @@ export class StripeProvider {
         dispatch(startWidgetLoading());
 
         if (this.reservation.amount > 0) {
-            const { id } = token;
-            stripe.confirmCardPayment(
-                this.reservation.payment_gateway_client_token, { payment_method: { card: { token: id } } }
+            const {id} = paymentMethod;
+            stripe.confirmPayment(
+                {
+                    elements, 
+                    clientSecret: this.reservation.payment_gateway_client_token, 
+                    confirmParams: {
+                        return_url: 'https://example.com',
+                        payment_method: paymentMethod.id
+                    },
+                    redirect: "if_required"
```

Comment:
why this is needed ?

Reply:
This prevents the user to get redirected to the `return_url` if the payment method is not a redirect-based method.

Reply:
but we dont have a return_url and plus its hardcoded i do think that we should remove both
* return_url
* redirect: "if_required"

Reply:
check doc https://stripe.com/docs/js/payment_intents/confirm_payment